nexus/src/app/background/sync_service_zone_nat.rs

```diff
@@ -21,6 +21,18 @@ use std::net::{IpAddr, SocketAddr};
 use std::num::NonZeroU32;
 use std::sync::Arc;
 
+// Minumum number of boundary NTP zones that should be present in a valid
+// set of service zone nat configurations.
+const MIN_NTP_COUNT: usize = 1;
+
+// Minumum number of nexus zones that should be present in a valid
+// set of service zone nat configurations.
+const MIN_NEXUS_COUNT: usize = 3;
```

Comment:
I'm not sure we should do this -- it's possible that one of the three Nexus instances have gone away, and we'd still want our NAT entries to be up-to-date.

This is a reasonable goal to try to achieve in the "graceful shutdown" case -- if we want three Nexus instances to run, we should provision a 4th one before removing one of the original 3 -- but if a sled is yanked from the rack, we **do** have two Nexus instances. That's just a truth! We should aspire to have the blueprint creator create a new Nexus service, provision it, and get the NAT entries populated, but it's **very possible** to run under our redundancy expectations. That's why we use redundancy!

Reply:
Would setting all of the minimums to `1` be a reasonable compromise? If we don't have at least 1 `Nexus`, `NTP`, and `ExternalDns` zone that can be found in inventory, I would think we have more serious problems, no?

Reply:
Chatting with some folks in the update sync today, it sounds like the inventory system -- which gathers the inventory in a "collection", which contains a lot of objects -- may, at any time, simply "not report a sled" within that collection. Theoretically, that means we could see an "inventory collection" that doesn't contain any sleds which contain Nexus, NTP, external DNS, etc.

This has some weird implications for depending on the inventory system as the source-of-truth here, but without a full implementation of blueprints, I acknowledge that there isn't a great alternative yet.

So: "Could we set all the minimums to 1?" That would stop us from propagating the state of the inventory system if we saw a blip that eliminated all of these critical zones. So in that sense, it's arguably better than not doing the check! I also think it avoids "breaking" NAT when we're under-provisioned, as I mentioned in the case below.

If we're on the same page that, eventually, the right source-of-truth is "info from the blueprint, somehow", I think this is a reasonable intermediate step.

Reply:
(also, I appreciate you tolerating all this churn. Getting RPWs + NAT propagation right is tricky, and having this portion of the system be "not-totally-ready" does make this extra hairy. Thank you for pushing through regardless)

Reply:
No worries! I know this is _critical_ so I appreciate the extra eyes and feedback!